crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1034,30 +1011,28 @@ impl<'w, 's, T: 'static> SystemParamFetch<'w, 's> for NonSendState<T> {
 pub struct OptionNonSendState<T>(NonSendState<T>);
 
 impl<'w, T: 'static> SystemParam for Option<NonSend<'w, T>> {
-    type Fetch = OptionNonSendState<T>;
+    type State = OptionNonSendState<T>;
 }
 
 // SAFETY: Only reads a single non-send resource
-unsafe impl<T: 'static> ReadOnlySystemParamFetch for OptionNonSendState<T> {}
+unsafe impl<T: 'static> ReadOnlySystemParamState for OptionNonSendState<T> {}
 
 // SAFETY: this impl defers to `NonSendState`, which initializes
 // and validates the correct world access
 unsafe impl<T: 'static> SystemParamState for OptionNonSendState<T> {
     fn init(world: &mut World, system_meta: &mut SystemMeta) -> Self {
         Self(NonSendState::init(world, system_meta))
     }
-}
 
-impl<'w, 's, T: 'static> SystemParamFetch<'w, 's> for OptionNonSendState<T> {
-    type Item = Option<NonSend<'w, T>>;
+    type Item<'w, 's> = Option<NonSend<'w, T>>;
```

Comment:
Style nit: move the associated type to the top of the impl block. Same with all of the others.